environments/servlet/core/src/main/java/org/jboss/weld/environment/jetty/JettyContainer.java

```diff
@@ -21,75 +21,45 @@
 
 import org.jboss.weld.environment.servlet.Container;
 import org.jboss.weld.environment.servlet.ContainerContext;
-import org.jboss.weld.environment.servlet.EnhancedListener;
 import org.jboss.weld.environment.servlet.logging.JettyLogger;
 import org.jboss.weld.resources.spi.ResourceLoader;
 
 /**
- * Jetty 7.2+, 8.x and 9.x container.
+ * Jetty &gt;= 9.4.20 container.
+ * <p>This container requires that the jetty server register DecoratingListener
+ * to dynamically register a decorator instance that wraps the {@link WeldDecorator}
+ * added as an attribute.   The jetty 'cdi' module does this and indicates it's
+ * availability by setting the "org.eclipse.jetty.cdi" attribute to "DecoratingListener"
+ * </p>
  *
- * @author <a href="mailto:ales.justin@jboss.org">Ales Justin</a>
+ * @see JettyLegacyContainer
+ * @author <a href="mailto:gregw@webtide.com">Greg Wilkins</a>
  */
 public class JettyContainer extends AbstractJettyContainer {
 
     public static final Container INSTANCE = new JettyContainer();
 
-    private static final String JETTY_SERVERNAME = "jetty";
-
-    private static final String JETTY_REQUIRED_CLASS_NAME = "org.eclipse.jetty.servlet.ServletHandler";
-
-    private static final int MAJOR_VERSION = 7;
-    private static final int MINOR_VERSION = 2;
-
     protected String classToCheck() {
-        // This is not used anyway - server implementation classes are hidden from the web application in Jetty by default
-        // Instead ServletContext.getServerInfo() is parsed
-        return JETTY_REQUIRED_CLASS_NAME;
+        // Never called because touch is overridden below.
+        throw new UnsupportedOperationException();
```

Comment:
I know this isn't supposed to be invoked and I am nitpicking, but should that ever accidentally happen, I'd prefer to either see dummy string returned or an exception with explanation.